synapse/models/person.py

```diff
@@ -47,8 +47,46 @@ def getModelDefs(self):
                 ('ps:contactlist', ('guid', {}), {
                     'doc': 'A GUID for a list of associated contacts.',
                 }),
+                ('ps:workhist', ('guid', {}), {
+                    'doc': "A GUID representing entry in a contact's work history.",
+                }),
             ),
             'forms': (
+                ('ps:workhist', {}, (
+                    ('contact', ('ps:contact', {}), {
+                        'doc': 'The contact which has the work history.',
+                    }),
+                    ('org', ('ou:org', {}), {
+                        'doc': 'The org that this work history orgname refers to.',
+                    }),
+                    ('orgname', ('ou:name', {}), {
+                        'doc': 'The reported name of the org the contact worked for.',
+                    }),
+                    ('orgfqdn', ('inet:fqdn', {}), {
+                        'doc': 'The reported fqdn of the org the contact worked for.',
+                    }),
+                    ('jobtype', ('ou:jobtype', {}), {
```

Comment:
Would this include type as in whether the role is PT/FTE/Contractor? (Also, would we want to be able to tie someone's work history to a specific contract?

Reply:
Designating the taxonomy hierarchy would be up to the users and power-up developers, but my thoughts were more about the type of work being done. I think we could probably add an enum about full-time vs part-time to deconflate that dimension.

Re contracts: Is this in regards to their employment contract or more like a gov "i worked on the foo contract" type thing?

Reply:
I was thinking more of the latter scenario for contracts (like if we were modeling Reign at VTX and wanted to show that she worked on wm and pg, how could we do that?)